lib/parse/blueprint.js

```diff
@@ -81,6 +94,28 @@ module.exports = function(filePath, autoOptions, routeMap) {
                     parseAction(parsedUrl, autoOptionsAction, routeMap);
                 });
             }
+
+            function getBasePath(result, basePathKey) {
+                return result.ast.metadata.reduce(function (basePath, item) {
+                    if (item.name.toLowerCase() === basePathKey.toLowerCase()) {
+                        return '/' + item.value.replace(/^\/|\/$/g, '');
+                    }
+                    return basePath;
+                }, '');
+            }
+
+            function updateWithBasePath(basePath, routeMap) {
+                if (basePath !== '') {
```

Comment:
Could this possibly be checked before making this function call? Say in the initial check on L:38 ?

Reply:
Good point-- it's also a redundant check because line 38 covers this case.